backend/fms_core/viewsets/_fetch_data.py

```diff
@@ -249,8 +249,26 @@ def fetch_export_data(self, ids: List[int] =[]) -> Tuple[List, int]:
 
                                                                SELECT sample.id AS id, full_location FROM container_hierarchy JOIN fms_core_sample AS sample ON sample.container_id=container_hierarchy.id 
                                                                WHERE sample.id IN  %s;''', params=[sample_ids])
+            
+            samples_with_site = Sample.objects.raw('''WITH RECURSIVE container_hierarchy(id, parent, coordinate_id, site_name) AS (
+                                                      SELECT container.id, container.location_id, coordinate.id, container.barcode::varchar
+                                                      FROM fms_core_container AS container
+                                                      LEFT OUTER JOIN fms_core_coordinate AS coordinate ON container.coordinate_id=coordinate.id
+                                                      WHERE (container.location_id IS null and container.kind='site')
+
+                                                      UNION ALL
+
+                                                      SELECT container.id, container.location_id, coordinate.id, container_hierarchy.site_name
+                                                      FROM container_hierarchy
+                                                      JOIN fms_core_container AS container ON container_hierarchy.id=container.location_id
+                                                      LEFT OUTER JOIN fms_core_coordinate AS coordinate ON container.coordinate_id=coordinate.id
+                                                      )
+
+                                                      SELECT sample.id AS id, site_name FROM container_hierarchy JOIN fms_core_sample AS sample ON sample.container_id=container_hierarchy.id 
+                                                      WHERE sample.id IN  %s;''', params=[sample_ids])
 
         location_by_sample = {sample.id: sample.full_location for sample in samples_with_full_location }
+        site_by_sample = {sample.id: sample.site_name for sample in samples_with_site }
```

Comment:
My IDE (i.e. Emacs) warns me that `samples_with_site` is possible unbound.

Reply:
yeah not at that point though ... i will test it below when assigning to site.

Reply:
fixed ...